src/System.Collections/tests/Generic/Queue/QueueTests.cs

```diff
@@ -31,6 +32,106 @@ public void Ctors()
         }
 
         [Theory]
+        [MemberData("Collections")]
+        public void Ctor_IEnumerable(IEnumerable<int> collection)
+        {
+            var q = new Queue<int>(collection);
+
+            Assert.Equal(collection, q);
+
+            foreach (int item in collection)
+            {
+                Assert.Equal(item, q.Dequeue());
+            }
+        }
+
+        [Theory]
+        [MemberData("Collections")]
+        public void Ctor_IEnumerable_EnqueueDequeue(IEnumerable<int> collection)
+        {
+            var q = new Queue<int>(collection);
+            q.Enqueue(int.MaxValue);
+            q.Enqueue(int.MinValue);
+
+            foreach (int item in collection)
+            {
+                Assert.Equal(item, q.Dequeue());
+            }
+            Assert.Equal(int.MaxValue, q.Dequeue());
+            Assert.Equal(int.MinValue, q.Dequeue());
+        }
+
+        [Theory]
+        [MemberData("NonEmptyCollections")]
+        public void Ctor_IEnumerable_Dequeue(IEnumerable<int> collection)
+        {
+            var q = new Queue<int>(collection);
+            IEnumerator<int> e = collection.GetEnumerator();
+
+            e.MoveNext();
+            q.Dequeue();
```

Comment:
Or just a `foreach`? If those are equal then why an explicit `MoveNext()` to special-case the first item?